cli/cmd/policies.go

```diff
@@ -37,53 +32,28 @@ func policiesFromKubeResources(yamlPaths []string) ([]deployment, error) {
+		objUnstructured := objs[0]
+
+		name := objUnstructured.GetName()
+		namespace := orDefault(objUnstructured.GetNamespace(), "default")
+		gvk := objUnstructured.GetObjectKind().GroupVersionKind()
 
 		var annotation string
 		var workloadSecretID string
 		var role manifest.Role
-		switch obj := objAny.(type) {
-		case *kubeapi.Pod:
-			annotation = obj.Annotations[initdata.InitdataAnnotationKey]
-			role = manifest.Role(obj.Annotations[contrastRoleAnnotationKey])
-			workloadSecretID = obj.Annotations[workloadSecretIDAnnotationKey]
-		case *kubeapi.Deployment:
-			annotation = obj.Spec.Template.Annotations[initdata.InitdataAnnotationKey]
-			role = manifest.Role(obj.Spec.Template.Annotations[contrastRoleAnnotationKey])
-			workloadSecretID = obj.Spec.Template.Annotations[workloadSecretIDAnnotationKey]
-		case *kubeapi.ReplicaSet:
-			annotation = obj.Spec.Template.Annotations[initdata.InitdataAnnotationKey]
-			role = manifest.Role(obj.Spec.Template.Annotations[contrastRoleAnnotationKey])
-			workloadSecretID = obj.Spec.Template.Annotations[workloadSecretIDAnnotationKey]
-		case *kubeapi.StatefulSet:
-			annotation = obj.Spec.Template.Annotations[initdata.InitdataAnnotationKey]
-			role = manifest.Role(obj.Spec.Template.Annotations[contrastRoleAnnotationKey])
-			workloadSecretID = obj.Spec.Template.Annotations[workloadSecretIDAnnotationKey]
-		case *kubeapi.DaemonSet:
-			annotation = obj.Spec.Template.Annotations[initdata.InitdataAnnotationKey]
-			role = manifest.Role(obj.Spec.Template.Annotations[contrastRoleAnnotationKey])
-			workloadSecretID = obj.Spec.Template.Annotations[workloadSecretIDAnnotationKey]
-		case *kubeapi.Job:
-			annotation = obj.Spec.Template.Annotations[initdata.InitdataAnnotationKey]
-			role = manifest.Role(obj.Spec.Template.Annotations[contrastRoleAnnotationKey])
-			workloadSecretID = obj.Spec.Template.Annotations[workloadSecretIDAnnotationKey]
-		case *kubeapi.CronJob:
-			name = obj.Name
-			annotation = obj.Spec.JobTemplate.Spec.Template.Annotations[initdata.InitdataAnnotationKey]
-			role = manifest.Role(obj.Spec.JobTemplate.Spec.Template.Annotations[contrastRoleAnnotationKey])
-			workloadSecretID = obj.Spec.JobTemplate.Spec.Template.Annotations[workloadSecretIDAnnotationKey]
-		case *kubeapi.ReplicationController:
-			name = obj.Name
-			annotation = obj.Spec.Template.Annotations[initdata.InitdataAnnotationKey]
-			role = manifest.Role(obj.Spec.Template.Annotations[contrastRoleAnnotationKey])
-			workloadSecretID = obj.Spec.Template.Annotations[workloadSecretIDAnnotationKey]
-		}
+		kuberesource.MapPodSpecWithMeta(objAny, func(meta *applymetav1.ObjectMetaApplyConfiguration, spec *applycorev1.PodSpecApplyConfiguration) (*applymetav1.ObjectMetaApplyConfiguration, *applycorev1.PodSpecApplyConfiguration) {
+			annotation = meta.Annotations[initdata.InitdataAnnotationKey]
+			role = manifest.Role(meta.Annotations[contrastRoleAnnotationKey])
+			workloadSecretID = meta.Annotations[workloadSecretIDAnnotationKey]
```

Comment:
Can we be certain that `meta != nil` at this point? Looks like we did not check before, but now it would be easy to add the check.